src/services/keyboard-control/keyboard-control.service.ts

```diff
@@ -0,0 +1,211 @@
+   * Sets the current map keyboard control configuration.
+   * The configuration is a key mapping between a key string and a KeyboardControlDefinition:
+   * - `action` is a predefine action from `KeyboardAction` enum, or a custom method:
+   * `(camera, scene, params, key) => boolean | void` - returning false will cancel the current keydown.
+   * - `validation` is a method that validates if the event should occur or not (`camera, scene, params, key`)
+   * - `params` is an object (or function that returns object) that will be passed into the action executor.
+   * @param {KeyboardControlDefinition} definitions
+   * @param {Function} keyMappingFn - Mapping function for custom keys
+   */
+  setKeyboardControls(definitions: KeyboardControlDefinition, keyMappingFn?: (keyEvent: KeyboardEvent) => string) {
+    if (!definitions) {
+      return this.removeKeyboardControls();
+    }
+
+    if (!this._currentDefinitions) {
+      this.registerEvents();
+    }
+
+    this._currentDefinitions = definitions;
+    this._keyMappingFn = keyMappingFn || this.defaultKeyMappingFn;
+
+    Object.keys(this._currentDefinitions).forEach(key => {
+      this._activeDefinitions[key] = null;
+    });
+  }
+
+  /**
+   * Removes the current set of keyboard control items, and unregister from map events.
+   */
+  removeKeyboardControls() {
+    this.unregisterEvents();
+    this._currentDefinitions = null;
+  }
+
+  private getAction(char: string): KeyboardControlParams {
+    return this._currentDefinitions[char] || null;
+  }
+
+  private defaultKeyMappingFn(keyEvent: KeyboardEvent): string {
+    return String.fromCharCode(keyEvent.keyCode);
+  }
+
+  private handleKeydown(e: KeyboardEvent) {
+    const char = this._keyMappingFn(e);
+    const action = this.getAction(char);
+
+    if (action) {
+      this._activeDefinitions[char] = action;
+    }
+  }
+
+  private handleKeyup(e: KeyboardEvent) {
+    const char = this._keyMappingFn(e);
+    const action = this.getAction(char);
+
+    if (action) {
+      this._activeDefinitions[char] = null;
+    }
+  }
+
+  private handleTick() {
+    const activeKeys = Object.keys(this._activeDefinitions);
+
+    activeKeys.forEach(key => {
+      const action = this._activeDefinitions[key];
+
+      if (action !== null) {
+        this.executeAction(action, key);
+      }
+    });
+  }
+
+  private getParams(paramsDef, camera, scene) {
+    if (!paramsDef) {
+      return {};
+    }
+
+    if (typeof paramsDef === 'function') {
+      return paramsDef(camera, scene);
+    }
+
+    return paramsDef;
+  }
+
+  private executeAction(execution: KeyboardControlParams, key: string) {
+    if (!this._currentDefinitions) {
+      return;
+    }
+
+    let execute = true;
+    const camera = this._viewer.camera;
+    const params = this.getParams(execution.params, camera, this._scene);
+
+    if (execution.validation) {
+      execute = execution.validation(camera, this._scene, params, key);
+    }
+
+    if (execute === true) {
```

Comment:
I used `===` because I want to make sure that the value is `true` and nothing else. 
Do you think it should be **any true value**?